### PR TITLE
Fixing hive partition getQualifiedName, adding resolver ending point

### PR DESCRIPTION
**Pull request by zhljen**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -60,3 +60,7 @@ servo_version=0.8.3
 spock_version=1.0-groovy-2.4
 swagger_version=1.3.12
 testing_mysql_server_version=0.1
+
+## speed up the build process
```
Comment (tgianos) [inline gradle.properties]: does this work? cool

File: metacat-common/src/main/java/com/netflix/metacat/common/dto/ResolveByUriResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import java.util.List;
+
+/**
+ * ResolveByUriResponseDto.
+ *
+ * @author zhenl
+ * @since 1.0.0
+ */
+@Data
```
Comment (tgianos) [inline metacat-common/src/main/java/com/netflix/metacat/common/dto/ResolveByUriResponseDto.java]: if this is a DTO please make the variables final. That way lombok won't create setters.

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy [hunk 3]
```diff
@@ -519,6 +523,12 @@ class MetacatFunctionalSpec extends Specification {
         database.tables.contains(tableName)
 
         when:
+        def resovlerRep = resolverApi.resolveByUri(false, resolveByUridto)
```
Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy]: Please add a test case with prefixSearch=true

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy [hunk 4]
```diff
@@ -788,6 +802,16 @@ class MetacatFunctionalSpec extends Specification {
         response.updated.empty
 
         when:
+        def resovlerRep = resolverApi.resolveByUri(false, resolveByUridto)
```
Comment (ajoymajumdar) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy]: Same here. Please add a test case with prefixSearch=true

File: gradle.properties (hunk 1 above)
Comment (zhljen) [inline gradle.properties]: Yes, it works!

File: metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy (hunk 4 above)
Comment (zhljen) [inline metacat-functional-tests/src/functionalTest/groovy/com/netflix/metacat/MetacatFunctionalSpec.groovy]: Added the test.

**Pull request merged**